File: language-support/js/daml-ledger/src/mapping/submit_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as mapping from '.';
import * as ledger from '..';
import * as grpc from 'daml-grpc';

export const SubmitRequest: mapping.Mapping<grpc.SubmitRequest, ledger.SubmitRequest> = {
    toObject(request: grpc.SubmitRequest): ledger.SubmitRequest {
        return {
            commands: mapping.Commands.toObject(request.getCommands()!)
        };
    },
    toMessage(request: ledger.SubmitRequest): grpc.SubmitRequest {
        const result = new grpc.SubmitRequest();
        result.setCommands(mapping.Commands.toMessage(request.commands));
        return result;
    }
}

File: sdk/canton/community/bindings-java/src/main/java/com/daml/ledger/javaapi/data/SubmitRequest.java
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.javaapi.data;

import com.daml.ledger.api.v2.CommandSubmissionServiceOuterClass;
import org.checkerframework.checker.nullness.qual.NonNull;

public final class SubmitRequest {

  public static CommandSubmissionServiceOuterClass.SubmitRequest toProto(
      @NonNull CommandsSubmission submission) {
    return CommandSubmissionServiceOuterClass.SubmitRequest.newBuilder()
        .setCommands(submission.toProto())
        .build();
  }

  public static CommandsSubmission fromProto(
      CommandSubmissionServiceOuterClass.@NonNull SubmitRequest request) {
    return CommandsSubmission.fromProto(request.getCommands());
  }
}
